### PR TITLE
Cleanup Dependencies, put everything possible inside of typed extensions

### PR DESCRIPTION
**Pull request by JacisNonsense**
- Moves `getTeamNumber()` and the like to an extension named `frc` (`frc.getTeamNumber()`)
- Moves library declarations to `wpi.deps`
```gradle
    compile wpi.deps.wpilib()
    compile wpi.deps.vendor.java()
    nativeZip wpi.deps.vendor.jni(wpi.platforms.roborio)
    nativeDesktopZip wpi.deps.vendor.jni(wpi.platforms.desktop)
```

- Cleans up internal declaration of libraries (including native)

### CODE REVIEW

File: src/main/groovy/edu/wpi/first/gradlerio/frc/FRCExtension.groovy [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            return teamDefault
+        return number
+    }
+
+    int getTeamNumber() {
+        if (project.hasProperty("teamNumber"))
+            return Integer.parseInt(project.property("teamNumber") as String)
+        def number = getTeamNumberFromJSON()
+        if (number < 0)
+            throw new TeamNumberNotFoundException()
```
Comment (JLLeitschuh) [inline src/main/groovy/edu/wpi/first/gradlerio/frc/FRCExtension.groovy]: I'm assuming this prints what a team needs to use to fix this?

File: src/main/groovy/edu/wpi/first/gradlerio/wpi/dependencies/WPIDepsExtension.groovy [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    List<String> wpilibJars() {
+        return ["edu.wpi.first.wpilibj:wpilibj-java:${wpi.wpilibVersion}".toString(),
+                "edu.wpi.first.ntcore:ntcore-java:${wpi.wpilibVersion}".toString(),
+                "edu.wpi.first.wpiutil:wpiutil-java:${wpi.wpilibVersion}".toString(),
+                "edu.wpi.first.thirdparty.frc2019.opencv:opencv-java:${wpi.opencvVersion}".toString(),
+                "edu.wpi.first.cscore:cscore-java:${wpi.wpilibVersion}".toString(),
+                "edu.wpi.first.cameraserver:cameraserver-java:${wpi.wpilibVersion}".toString(),
+                "edu.wpi.first.hal:hal-java:${wpi.wpilibVersion}".toString()]
```
Comment (JLLeitschuh) [inline src/main/groovy/edu/wpi/first/gradlerio/wpi/dependencies/WPIDepsExtension.groovy]: You could simplify a lot of this with a `[].map { it.toString() }`

Comment (JacisNonsense) [reply src/main/groovy/edu/wpi/first/gradlerio/wpi/dependencies/WPIDepsExtension.groovy]: It's not worth the extra iteration IMO

**Comment by JLLeitschuh** [comment]
Nice job with this BTW, this begins to help make the Kotlin DSL usable for teams.

**Pull request merged**